package-res/resources/web/pentaho/type/changes/Changeset.js

```diff
@@ -0,0 +1,187 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+define([
+  "./Change",
+  "./ListChange",
+  "./SimpleChange",
+  "../../util/object",
+  "../../util/error"
+], function(Change, ListChange, SimpleChange, O, error) {
+  "use strict";
+
+  /**
+   * @name Changeset
+   * @memberOf pentaho.type.changes
+   * @class
+   * @extends pentaho.type.changes.Change
+   * @amd pentaho/type/changes/Changeset
+   *
+   * @classDesc Class that represents a collection of changes in a set of properties
+   * in a [complex]{@linkplain pentaho.type.Complex} object.
+   *
+   * @constructor
+   * @description Creates a new instance.
+   *
+   * @param {!pentaho.type.Complex} owner - The [complex]{@linkplain pentaho.type.Complex} associated with this change.
+   */
+  return Change.extend("pentaho.type.changes.Changeset", /** @lends pentaho.type.changes.Changeset#*/{
+
+    constructor: function(owner) {
+      if(!owner) throw error.argRequired("owner");
+      this.base(owner);
+
+      this._properties = {};
+    },
+
+    //region public interface
+    /**
+     * @inheritdoc
+     */
+    get type() {
+      return "changeset";
+    },
+
+    /**
+     * Determines if this changeset contains the specified property.
+     *
+     * @param {nonEmptyString|!pentaho.type.Property.Type} name - The property name or type object.
+     *
+     * @return {boolean} `true` if the property exists and is referenced in this changeset, `false` otherwise.
+     */
+    has: function(name) {
+      var pName = this.owner.type.get(name).name;
+      return pName && O.hasOwn(this._properties, pName);
+    },
+
+    /**
+     * Gets an array with all of the property names contained in this changeset.
+     *
+     * @type !string[]
+     * @readonly
+     */
+    get propertyNames() {
+      return Object.keys(this._properties);
+    },
+
+    /**
+     * Asserts if this changesets contains any defined changes.
+     *
+     * @returns {boolean} `true` if at least one property change is defined, `false` otherwise.
```

Comment:
suggested wording:

'true' is at least one property change is defined, 'false' if no property changes are defined.